innovation_sweet_spots/getters/inputs.py

```diff
@@ -115,10 +121,49 @@ def get_cb_data(fpath=CB_PATH, cb_data_spec_path=CB_DATA_SPEC_PATH, use_cached=T
     return tables
 
 
+def download_hansard_data(fpath=HANSARD_PATH):
+    """
+    Downloads version 3.1.0 of the Hansard Speeches dataset.
+    Find more information about the dataset here: https://zenodo.org/record/4843485
+
+    Function can be used from command line as follows:
+        python -c "from innovation_sweet_spots.getters.inputs import download_hansard_data; download_hansard_data();"
+
+    Parameters
+    ----------
+    fpath : str
+        Location on disk for saving the data
+    """
+    # Download from Zenodo
+    logging.info(f"Collection and storing of Hansard data in progress")
+    zip_files = []
+    for filename, url in zip(ZENODO_FILES, ZENODO_URLS):
+        logging.info(f"Collecting {filename}")
+        filepath = fpath / filename
+        urllib.request.urlretrieve(url, fpath / filename)
```

Comment:
`urlretrieve` is kind of old hat (https://docs.python.org/3/library/urllib.request.html#urllib.request.urlretrieve) and will probably become deprecated

Reply:
It's not necessary to temporarily manage files on disk: i'd suggest streaming the data to a temp file and then only extracting the file you need. Also I'd replace `urlretrieve` with a streamed request to give more stability for large files. Something like this should do it (not tested, probably needs a little tweak on the filenames or something)

```python
import requests
from tempfile import TemporaryFile
import shutil
from zipfile import ZipFile


def stream_to_file(url, fileobj):
    """
    Stream a large file from a URL to a file object in a memory-efficient
    rate-efficient manner.
    """
    with requests.get(url, stream=True) as r:
        r.raise_for_status()
        shutil.copyfileobj(r.raw, fileobj)
    fileobj.seek(0)  # Reset the file pointer, ready for reading

def download_hansard_data(fpath=HANSARD_PATH):
    for filename, url in zip(ZENODO_FILES, ZENODO_URLS):
        with TemporaryFile() as fileobj:
            stream_to_file(url, fileobj)
            with ZipFile(fileobj) as zf:
                zf.extract(member=filename, path=fpath)
```

Reply:
EDIT: I switched `TemporaryFile` to `BytesIO`: even less writing to disk

Reply:
EDIT: I saw that you also need a mode for the json file, so extended the functionality a little